### PR TITLE
Set change subaddress index to u64::MAX - 1, update sample paykit

### PR DESCRIPTION
**Pull request by garbageslam**
This change is described in issue #1787

### CODE REVIEW

**Review by remoun — APPROVED** [review]

**Review by voloshyn — APPROVED** [review]
LGTM

**Review by iamalwaysuncomfortable — APPROVED** [review]
Approval modulo questions asking about why we are making this change/why it's better

File: account-keys/src/account_keys.rs [hunk 1]
```diff
@@ -35,8 +35,12 @@ use zeroize::Zeroize;
...
 pub const DEFAULT_SUBADDRESS_INDEX: u64 = 0;
 
-/// An account's "change address" is its 1st subaddress.
-pub const CHANGE_SUBADDRESS_INDEX: u64 = 1;
+/// u64::MAX is a reserved subaddress value for "invalid/none" (MCIP #36)
+pub const INVALID_SUBADDRESS_INDEX: u64 = u64::MAX;
+
+/// An account's "change address" is the 1st reserved subaddress,
+/// counting down from `u64::MAX`. (See MCIP #4, MCIP #36)
+pub const CHANGE_SUBADDRESS_INDEX: u64 = u64::MAX - 1;
```
[the remark this answers is not in the document]
Comment (garbageslam) [reply account-keys/src/account_keys.rs:43]: Please see MCIP #36 discussion

[the remark this answers is not in the document]
Comment (garbageslam) [reply account-keys/src/account_keys.rs:43]: Also discord

**Review by jcape — APPROVED** [review]
LGTM, though we may want to make the "high" subaddress range more definite in the future.

**Pull request merged**